src/IonEvent.ts

```diff
@@ -0,0 +1,473 @@
+
+    }
+    valueEquals(expected : IonTimestampEvent) : boolean {
+        if(expected.constructor.name !== IonTimestampEvent.name) return false;
+        return this.ionValue.equals(expected.ionValue);
+    }
+    writeIonValue(writer : Writer) : void {
+        writer.writeTimestamp(this.ionValue);
+    }
+}
+
+export class IonBlobEvent extends IonEvent {
+    constructor(eventType : IonEventType, ionType : IonType, fieldName : string, annotations : string[], depth : number, ionValue : Timestamp){
+        super(eventType , ionType, fieldName, annotations, depth, ionValue);
+    }
+    valueEquals(expected : IonBlobEvent) : boolean {
+        return this.ionValue === expected.ionValue;
+    }
+    writeIonValue(writer : Writer) : void {
+        let tempBuf = [];
+        for(let i = 0; i < this.ionValue.length; i++){
+            tempBuf.push(this.ionValue.charCodeAt(i))
+        }
+        writer.writeBlob(tempBuf);
+    }
+}
+export class IonClobEvent extends IonEvent {
+    constructor(eventType : IonEventType, ionType : IonType, fieldName : string, annotations : string[], depth : number, ionValue : Timestamp){
+        super(eventType , ionType, fieldName, annotations, depth, ionValue);
+    }
+    valueEquals(expected : IonClobEvent) : boolean {
+        return this.ionValue === expected.ionValue;
+    }
+    writeIonValue(writer : Writer) : void {
+        let tempBuf = [];
+        for(let i = 0; i < this.ionValue.length; i++){
+            tempBuf.push(this.ionValue.charCodeAt(i))
+        }
+        writer.writeClob(tempBuf);
+    }
+}
+/*
+
+Container events encapsulate other IonEvents
+
+ */
+export abstract class IonContainerEvent extends IonEvent {
+    events : IonEvent[];
+    constructor(eventType : IonEventType, ionType : IonType, fieldName : string, annotations : string[], depth : number, ionValue : any){
+        super(eventType , ionType, fieldName, annotations, depth, ionValue);
+    }
+
+    abstract valueEquals(expected : IonContainerEvent);
+
+    setEvents(inputEvents : IonEvent[]) {
+        this.events = inputEvents;
+    }
+
+    writeIonValue(writer : Writer) : void {
+        writer.writeNull(TypeCodes.NULL);
+    }
+}
+
+export class IonStructEvent extends IonContainerEvent {//no embed support as of yet.
+    constructor(eventType : IonEventType, ionType : IonType, fieldName : string, annotations : string[], depth : number, ionValue : any){
+        super(eventType , ionType, fieldName, annotations, depth, ionValue);
+
+    }
+    //two way equivalence between the structEvents, they must have the exact same number of equivalent elements.
+    valueEquals(expected : IonStructEvent) : boolean {
+        if(expected.constructor.name !== IonStructEvent.name) return false;
+        return this.structsEqual(this.events, expected.events) && this.structsEqual(expected.events, this.events);
+    }
+
+    //for each actual ionEvent, searches for an equivalent expected ionEvent,
```

Comment:
basically structs equal tells you that for each element in my (not set) there is an equal element in the other (not set), you need this to be a two way street since one (not set) could have two equal elements and the other could have only one.